source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -156,7 +157,14 @@ Http::FilterHeadersStatus CompressorFilter::decodeHeaders(Http::RequestHeaderMap
   }
 
   const auto& request_config = config_->requestDirectionConfig();
-  if (!end_stream && request_config.compressionEnabled() &&
+  // temp variable to preserve old behavior w/ http upgrade. should be removed when
+  // enable_compression_without_content_length_header runtime variable would be removed
+  const bool is_upgrade_allowed =
+      (!Http::Utility::isUpgrade(headers) && !Http::Utility::isH2UpgradeRequest(headers)) ||
```

Comment:
hm, I'd think this should be isUpgradeRequest, and I'm wondering why the tests are working correctly as is.  Can you take a look and make sure we get the right function here and have regression tests that fail with things as they are?

Reply:
agree. will rework naming. should be "is_not_upgrade" . but yes. the logic is correct. w/o this line integration tests for upgrade are failing with:
```
Expected: header map equal ignore order 0x7ffde7a7cf48
  Actual: 0x7ffde7a7ce78 (of type Envoy::Http::TestResponseHeaderMapImpl*), 
==============================Expected header map:==============================
':status', '101'
'connection', 'upgrade'
'upgrade', 'foo'
=======================is not equal to actual header map:=======================
':status', '101'
'vary', 'Accept-Encoding'
'upgrade', 'foo'
'connection', 'upgrade'
================================================================================

Stack trace:
```

aka envoy is adding "vary" header to signal that potentially (if accept-encoding was set) it could compress such request/response. (and from my understanding of initial reported issue  -that was the actuall issue. that browser was not expecting this header. nothing was actually compressed (as i cannot see from code how that could happen if accept-encoding was not set in request)).

Reply:
The HTTP/2 codec is supposed to transform HTTP/2 style upgrades to HTTP/1 style for the lifetime of the request in Envoy, so it should be impossible to look like an H2 upgrade request at this point in the pipeline.  Do you have an integration test where isH2UpgradeRequest is true?  If so I'll download the patch and sort out why my assumptions are off :-P

Reply:
nope. was just saw it in utility and just added as well. was not aware that it will make h2 looks like h1 in codec. will remove